firebase-common/src/main/java/com/google/firebase/emulators/EmulatedServiceSettings.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.emulators;
+
+import androidx.annotation.NonNull;
+
+/**
+ * Settings to connect a single Firebase service to a local emulator.
+ *
+ * <p>TODO(samstern): Un-hide this once Firestore, Database, and Functions are implemented
+ *
+ * @see EmulatorSettings
+ * @hide
+ */
+public class EmulatedServiceSettings {
+
+  public static final class Builder {
+
+    private final String host;
+    private final int port;
+
+    /**
+     * Create a new EmulatedServiceSettings builder.
+     *
+     * @param host the host where the local emulator is running. If you want to access 'localhost'
+     *     from an Android Emulator use '10.0.2.2' instead.
+     * @param port the port where the local emulator is running.
+     */
+    public Builder(@NonNull String host, int port) {
```

Comment:
I've always been told that required arguments should be in the `Builder` constructor.  The awkward thing here is that the `Builder` has no other methods at all but in the API discussion it was clear that we may want to extend this object one day (`setInstance()` maybe or `setCredentials` as you suggest below) so having a `Builder` now rather than a plain constructor is more future proof.

Reply:
+1 on required parameters as builder constructor parameters

Although to the point of being future-proof: it seems that it's possible to be future-proof by not having a builder at all at this point and expose the constructor directly, then add the builder when(and most importantly **if**) needed and tell people to use the builder for additional functionality. wdyt?

Reply:
I am open to whatever you think.  I guess let's do the shorter thing now and do the builder later.

Reply:
Builder removed.

Reply:
Whatever we decide has to go through api review, has it happened yet btw?

Reply:
Yes the API (go/firebase-emulator-connection-api) was approved before I started this PR.

IMO we don't have to go back through API review to change a Builder to a non-Builder, especially since you are the Android platform approver.

Reply:
Just to clarify, I don't have a vote in the API council, I just offer my recommendations :)

That said, it would not hurt to drop an email to your api review thread, but it's definitely up to you.